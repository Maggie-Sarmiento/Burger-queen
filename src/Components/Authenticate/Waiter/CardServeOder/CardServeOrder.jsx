/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/prop-types */
import {
  createTheme,
  Grid,
  Paper,
  ThemeProvider,
  Typography,
} from '@mui/material';
import React from 'react';
import Box from '@mui/material/Box';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
// import MoreVertIcon from '@mui/icons-material/MoreVert';
import Checkbox from '@mui/material/Checkbox';
import styles from '../../Admin/Menu/CardMenu.module.css';
import BtnAddCheckService from '../BtnAddCheckService/BtnAddCheckService';

const label = { inputProps: { 'aria-label': 'Checkbox demo' } };

const btnAdd = {
  display: 'flex',
  justifyContent: 'end',
  position: 'absolute',
  top: '-10px',
  right: '0',
  padding: '0',
};

const theme = createTheme({
  components: {
    MuiTypography: {
      variants: [
        {
          props: {
            variant: 'body2',
          },
          style: {
            fontSize: 11,
          },
        },
        {
          props: {
            variant: 'body3',
          },
          style: {
            fontSize: 9,
          },
        },
      ],
    },
  },
});

const CardServeOrder = ({
  order, id, refreshData, setRefreshData,
}) => {
  const calcTotalTime = (startTime, endTime) => Math.round((endTime - startTime) / 60000);
  return (
    <Grid item xs={12} sm={6}>
      <ThemeProvider theme={theme}>
        <Paper elevation={3} className="paper">
          <img src={order.image} alt="" className={styles.imgCard} />
          <Box
            sx={{
              paddingX: 5,
              paddingBottom: 2,
              position: 'relative',
            }}
          >
            <Box>
              <Box
                sx={{
                  display: 'flex',
                  alignItems: 'center',
                }}
                marginTop={0.5}
              />
              <Box sx={btnAdd}>
                <BtnAddCheckService
                  order={order}
                  id={id}
                  refreshData={refreshData}
                  setRefreshData={setRefreshData}
                  totalTime={calcTotalTime(order.startTime, order.endTime)}
                />
              </Box>
            </Box>
            <Typography
              variant="subtitle1"
              component="h2"
              className={styles.productName}
              sx={{
                alignSelf: 'center',
                fontWeight: 'bold',
                fontSize: '0.8em',
              }}
            >
              {order.client}
            </Typography>
            <Typography
              variant="subtitle1"
              component="h2"
              className={styles.productName}
              sx={{
                alignSelf: 'center',
                fontWeight: 'bold',
                fontSize: '0.8em',
              }}
            >
              Tiempo en completar pedido:&nbsp;
              {calcTotalTime(order.startTime, order.endTime)}
              &nbsp;minutos
            </Typography>
            <TableContainer>
              <Table>
                <TableHead>
                  <TableRow>
                    <TableCell><b>Cant.</b></TableCell>
                    <TableCell><b>Producto</b></TableCell>
                    <TableCell><b>Servido</b></TableCell>
                  </TableRow>
                </TableHead>
                <TableBody>
                  {order.products?.map((product) => (
                    <TableRow key={product.id}>
                      <TableCell>{product.id}</TableCell>
                      <TableCell>{product.name}</TableCell>
                      <TableCell>
                        <Checkbox {...label} color="success" />
                      </TableCell>
                    </TableRow>
                  ))}
                </TableBody>
              </Table>
            </TableContainer>
            <Typography
              variant="subtitle1"
              component="h2"
              className={styles.productName}
              sx={{
                alignSelf: 'center',
                fontWeight: 'bold',
                fontSize: '0.8em',
                paddingLeft: '1.2em',
              }}
            >
              Mesa:&nbsp;
              {order.table}
            </Typography>
          </Box>
        </Paper>
      </ThemeProvider>
    </Grid>
  );
};

export default CardServeOrder;
