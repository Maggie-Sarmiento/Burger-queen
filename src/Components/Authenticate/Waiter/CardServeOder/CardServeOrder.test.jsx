/* eslint-disable no-undef */
import React from 'react';
import { create, act } from 'react-test-renderer';
import CardServeOrder from './CardServeOrder';

describe('test CardServeOrder', () => {
  test('render CardServeOrder', () => {
    const mockOrder = jest.fn();
    let root;
    act(() => {
      root = create(
        <CardServeOrder order={mockOrder} />,
      );
    });
    expect(root.toJSON()).toMatchSnapshot();
  });
});
