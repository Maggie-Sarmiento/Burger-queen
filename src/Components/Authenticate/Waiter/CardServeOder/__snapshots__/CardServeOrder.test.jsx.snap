// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test CardServeOrder render CardServeOrder 1`] = `
<div
  className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 css-13i4rnv-MuiGrid-root"
>
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation3 paper css-3su884-MuiPaper-root"
  >
    <img
      alt=""
      className="imgCard"
    />
    <div
      className="MuiBox-root css-1px8riv"
    >
      <div
        className="MuiBox-root css-0"
      >
        <div
          className="MuiBox-root css-1yhm737"
        />
        <div
          className="MuiBox-root css-1f5i9a2"
        >
          <div
            className="css-f779bt-MuiStack-root"
          >
            <button
              className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-12ituje-MuiButtonBase-root-MuiButton-root"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="AddTaskIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M22 5.18 10.59 16.6l-4.24-4.24 1.41-1.41 2.83 2.83 10-10L22 5.18zM12 20c-4.41 0-8-3.59-8-8s3.59-8 8-8c1.57 0 3.04.46 4.28 1.25l1.45-1.45C16.1 2.67 14.13 2 12 2 6.48 2 2 6.48 2 12s4.48 10 10 10c1.73 0 3.36-.44 4.78-1.22l-1.5-1.5c-1 .46-2.11.72-3.28.72zm7-5h-3v2h3v3h2v-3h3v-2h-3v-3h-2v3z"
                />
              </svg>
              <span
                className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <h2
        className="MuiTypography-root MuiTypography-subtitle1 productName css-1tk5ht8-MuiTypography-root"
      />
      <h2
        className="MuiTypography-root MuiTypography-subtitle1 productName css-1tk5ht8-MuiTypography-root"
      >
        Tiempo en completar pedido: 
        NaN
         minutos
      </h2>
      <div
        className="MuiTableContainer-root css-rorn0c-MuiTableContainer-root"
      >
        <table
          className="MuiTable-root css-rqglhn-MuiTable-root"
          role={null}
        >
          <thead
            className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
            role={null}
          >
            <tr
              className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
              role={null}
            >
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                <b>
                  Cant.
                </b>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                <b>
                  Producto
                </b>
              </th>
              <th
                aria-sort={null}
                className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
                scope="col"
              >
                <b>
                  Servido
                </b>
              </th>
            </tr>
          </thead>
          <tbody
            className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
            role={null}
          />
        </table>
      </div>
      <h2
        className="MuiTypography-root MuiTypography-subtitle1 productName css-5gagl7-MuiTypography-root"
      >
        Mesa: 
      </h2>
    </div>
  </div>
</div>
`;
